fix bug in case the A or AAAA records is processed before SRV record

stefaneicher opened the pull request:
Hi Kay,

I had an error using the library. see ServiceInfoImplTest.java
the error occurs if  you uncomment this row
allAnswers = aRecordsLast(allAnswers);

Gruss Stefan

kaikreuzer commented on src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java:
```diff
@@ -0,0 +1,61 @@
+                new byte[]{});
+        jmDNS.addListener(serviceInfo, null);
+        jmDNS.handleResponse(msg);
+
```
You should close jmDNS here again.

kaikreuzer commented on src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java:
```diff
@@ -0,0 +1,61 @@
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author stefan eicher https://github.com/stefaneicher
```
Please just do `@author Stefan Eicher`

kaikreuzer commented on src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java:
```diff
@@ -0,0 +1,61 @@
+        assertArrayEquals(serviceInfo.getInet4Addresses()[0].getAddress(), new  byte[]{(byte) 192, (byte) 168,88, (byte) 236});
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
```
what is this good for? This gives a warning in my IDE...

stefaneicher replied:
I use Intellij ultimate and I want to handle all warnings the IDE gives me to let other peoble know I addressed them. Unfortentualy eclipse does not understand them.

![image](https://cloud.githubusercontent.com/assets/1171777/17244478/6005328a-5582-11e6-8b06-84e32e4c57a0.png)

![image](https://cloud.githubusercontent.com/assets/1171777/17244511/8d0e03c4-5582-11e6-98e3-fa5abcd72075.png)

kaikreuzer replied:
I see. How great - to make a warning disappear in one IDE adds one in another...
Well, fine for me to leave it as is.

kaikreuzer commented:
Thanks Stefan,

Unfortunately, your test fails for me:
```
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.011 sec <<< FAILURE! - in javax.jmdns.impl.ServiceInfoImplTest
test_ip_address_is_set(javax.jmdns.impl.ServiceInfoImplTest)  Time elapsed: 0.01 sec  <<< ERROR!
java.io.IOException: DNSIncoming corrupted message
	at javax.jmdns.impl.DNSIncoming.<init>(DNSIncoming.java:199)
	at javax.jmdns.impl.ServiceInfoImplTest.test_ip_address_is_set(ServiceInfoImplTest.java:21)
```
Any idea what could be wrong?

Please also make sure to [sign off your commit](https://github.com/jmdns/jmdns/blob/master/CONTRIBUTING.md#sign-your-work), which is required for contributions.

stefaneicher commented:
Sorry something went woring as I check in the binary file. It is updated now.
Also see the wiresahrk file a_record_before_srv.pcapng for more details.

![image](https://cloud.githubusercontent.com/assets/1171777/17245306/2ea5923a-5586-11e6-85b2-af00da96f42e.png)

kaikreuzer commented on src/main/java/javax/jmdns/impl/JmDNSImpl.java:
```diff
@@ -1429,6 +1412,48 @@ void handleResponse(DNSIncoming msg) throws IOException {
+     * DIST500_7-F07_OC030_05_03941.local: type NSEC, class IN, cache flush, next domain name DIST500_7-F07_OC030_05_03941.local
+     */
+    private List<DNSRecord> aRecordsLast(List<DNSRecord> allAnswers) {
+        ArrayList<DNSRecord> ret = new ArrayList<DNSRecord>();
```
To slightly optimize it, I would suggest to change it to
```
        List<DNSRecord> ret = new ArrayList<DNSRecord>(allAnswers.size());
```

kaikreuzer replied:
Did you see this comment?

kaikreuzer commented:
Hi Stefan,

Thanks for the updates, looks good to me now!
Before I can merge, may I ask you to squash your commits into one and [sign-off the resulting commit](https://github.com/jmdns/jmdns/blob/master/CONTRIBUTING.md#sign-your-work)? Many thanks!

stefaneicher commented:
Hallo Kai, sicher doch. I will do it on monday. In case you just merge my
commit I will try to make it even nicer (undo my intellij formatting...).
Gruss Stefan

Kai Kreuzer <notifications@github.com> schrieb am Fr., 29. Juli 2016 16:53:

> Hi Stefan,
>
> Thanks for the updates, looks good to me now!
> Before I can merge, may I ask you to squash your commits into one and sign-off
> the resulting commit
> <https://github.com/jmdns/jmdns/blob/master/CONTRIBUTING.md#sign-your-work>?
> Many thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jmdns/jmdns/pull/80#issuecomment-236202085>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABHhQReKBPEkQ0yQWtKkbTpWzuJ8P9u9ks5qahPSgaJpZM4JT7iK>
> .
>

kaikreuzer commented:
I will wait for your "go" and merge then as-is. Schönes Wochenende!

stefaneicher commented:
Hi kai, I squashed all to one commit. I would highly appreciate if you
could release a new version with this bug fix. Can you do that?

Kai Kreuzer <notifications@github.com> schrieb am Fr., 29. Juli 2016 17:02:

> I will wait for your "go" and merge then as-is. Schönes Wochenende!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jmdns/jmdns/pull/80#issuecomment-236204793>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABHhQQ_KnRS5IjIoCMi2lFqEsljU7Bq3ks5qahYYgaJpZM4JT7iK>
> .
>

kaikreuzer commented:
Hi Stefan,

One comment left (https://github.com/jmdns/jmdns/pull/80#discussion_r72804651) to answer and your commit is still not signed-off.
I will certainly do a new release with this fix, the question is when ;-)

stefaneicher commented:
Hi Kai,
I added the sing-of, removed the intellij files and added the
"List<DNSRecord> ret = new ArrayList<DNSRecord>(allAnswers.size());"

2016-08-02 10:59 GMT+02:00 Kai Kreuzer <notifications@github.com>:

> Hi Stefan,
>
> One comment left (#80 (comment)
> <https://github.com/jmdns/jmdns/pull/80#discussion_r72804651>) to answer
> and your commit is still not signed-off.
> I will certainly do a new release with this fix, the question is when ;-)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jmdns/jmdns/pull/80#issuecomment-236845494>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABHhQSL7brTEBXnbGNjQ2wRgoVQdrQatks5qbwbsgaJpZM4JT7iK>
> .
>

kaikreuzer commented:
Many thanks!

Now I have to ask you for a favor: Could you briefly review my PR https://github.com/jmdns/jmdns/pull/86 and comment if it is ok to merge?

stefaneicher commented:
Off course. I will today or if I can tomorrow

Kai Kreuzer <notifications@github.com> schrieb am Di., 2. Aug. 2016 17:42:

> Merged #80 <https://github.com/jmdns/jmdns/pull/80>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jmdns/jmdns/pull/80#event-742588067>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/ABHhQbPmkR1cnjU9xXyTF6UGLH2N3vmmks5qb2VpgaJpZM4JT7iK>
> .
>
